### ISSUE TITLE
Improve interface for train_from_viewer

### ISSUE THREAD

**Issue by AbigailMcGovern** [AbigailMcGovern/iterseg#4]
## Issue
Currently, the `train_from_viewer` function shows only whole volumes of training images and ground truth. The function generates a variety of output, some of which would be useful if visualised during the training process. This includes:

- training loss scores (both total and for each feature map)
- validation loss scores (both total and for each feature map)
- augmented image chunks used for training and validation
- ground truth feature maps used for training and validation

Furthermore, the user cannot investigate the data during the training process probably due to GIL. 

## Proposed solution

- Visualise some of the above output in the napari window with live updates during training.
- use multiprocessing or multithreading to instantiate a new thread/process to allow user to still use napari window whilst the training takes place in the background.
